File: android_claude/domain/src/main/java/com/zarabandajose/transcripto/domain/repository/CipherRepository.kt
package com.zarabandajose.transcripto.domain.repository

import com.zarabandajose.transcripto.core.common.Result
import com.zarabandajose.transcripto.domain.model.CipherParams

/**
 * Interfaz para operaciones de cifrado.
 * Define los contratos que deben implementar los diferentes métodos de cifrado.
 */
interface CipherRepository {
    
    /**
     * Cifra un texto usando los parámetros especificados.
     * 
     * @param input Texto a cifrar
     * @param params Parámetros de cifrado
     * @return Result con el texto cifrado o error
     */
    suspend fun encrypt(input: String, params: CipherParams): Result<String>
    
    /**
     * Descifra un texto usando los parámetros especificados.
     * 
     * @param input Texto a descifrar
     * @param params Parámetros de descifrado
     * @return Result con el texto descifrado o error
     */
    suspend fun decrypt(input: String, params: CipherParams): Result<String>
}

File: android_flutter_claude/transcripto/packages/domain/lib/src/repositories/cipher_repository.dart
import 'package:core_common/core_common.dart';
import '../entities/cipher_params.dart';

/// Abstract repository for cipher operations
abstract class CipherRepository {
  /// Encrypts text using the specified parameters
  Result<String> encrypt(String input, CipherParams params);
  
  /// Decrypts text using the specified parameters
  Result<String> decrypt(String input, CipherParams params);
}
